surveyutils/input.go

```diff
@@ -0,0 +1,221 @@
+package surveyutils
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"gopkg.in/AlecAivazis/survey.v1"
+	"gopkg.in/AlecAivazis/survey.v1/terminal"
+)
+
+func UseStdio(io terminal.Stdio) {
+	stdio = &io
+}
+
+var stdio *terminal.Stdio
+
+func AskOne(p survey.Prompt, response interface{}, v survey.Validator, opts ...survey.AskOpt) error {
+	if stdio != nil {
+		opts = append(opts, survey.WithStdio(stdio.In, stdio.Out, stdio.Err))
+	}
+	return survey.AskOne(p, response, v, opts...)
+}
+
+
+func GetYesInput(msg string) (bool, error) {
+	var yesAnswer string
+
+	if err := GetStringInputDefault(
+		msg,
+		&yesAnswer,
+		"N",
+	); err != nil {
+		return false, err
+	}
+	return strings.ToLower(yesAnswer) == "y", nil
+}
+
+func GetStringInput(msg string, value *string) error {
+	return GetStringInputDefault(msg, value, "")
+}
+
+func GetStringInputDefault(msg string, value *string, defaultValue string) error {
+	prompt := &survey.Input{Message: msg, Default: defaultValue}
+	if err := AskOne(prompt, value, nil); err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetUint32Input(msg string, value *uint32) error {
+	return GetUint32InputDefault(msg, value, 0)
+}
+
+func GetUint32InputDefault(msg string, value *uint32, defaultValue uint32) error {
+	var strValue string
+	prompt := &survey.Input{Message: msg, Default: strconv.Itoa(int(defaultValue))}
+	if err := AskOne(prompt, &strValue, nil); err != nil {
+		return err
+	}
+	val, err := strconv.Atoi(strValue)
+	if err != nil {
+		return err
+	}
+	*value = uint32(val)
+	return nil
+}
+
+func GetBoolInput(msg string, value *bool) error {
+	return GetBoolInputDefault(msg, value, false)
+}
+
+func GetBoolInputDefault(msg string, value *bool, defaultValue bool) error {
+	var strValue string
+	defaultValueStr := "N"
+	if defaultValue {
+		defaultValueStr = "y"
+	}
+	prompt := &survey.Input{Message: msg + " [y/N]: ", Default: defaultValueStr}
```

Comment:
upper case N implies that it is the default. make it lower case?
or make the y upper case?